closed/src/java.base/share/classes/sun/security/provider/NativeDigest.java

```diff
@@ -161,18 +180,13 @@ synchronized protected final int engineDigest(byte[] out, int ofs, int len)
 
     synchronized public Object clone() throws CloneNotSupportedException {
         NativeDigest copy = (NativeDigest) super.clone();
```

Comment:
I don't think using `super.clone()` is appropriate here: it will yield state that may be inconsistent with the newly allocated `context`. I suggest it should just be:
```java
    return new NativeDigest(algorithm, digestLength, algIndx);
```

Reply:
that won't work, in the constructor `nativeCrypto.DigestCreateContext(0, algIndx)` is used while in clone() `nativeCrypto.DigestCreateContext(context, algIndx)` is needed to have openssl copy the current state of the cipher to the cloned cipher.

How about create an internal initCipher() method that will also take in the context and create an empty constructor. In the current constructor have initCipher() method called with context = 0 and in clone() invoke the empty constructor and then directly call initCipher() with the relevant context?

Reply:
Sorry, I missed the difference in the call to `DigestCreateContext` in `clone()` versus the constructor.
The only remaining change I request in `clone()` is to fix the indentation (no tabs in this repo).